tpu-client-next/src/quic_networking/quic_client_certificate.rs

```diff
@@ -10,6 +10,14 @@ pub struct QuicClientCertificate {
 }
 
 impl QuicClientCertificate {
+    pub fn with_option(keypair: Option<&Keypair>) -> Self {
+        if let Some(keypair) = keypair {
+            QuicClientCertificate::new(keypair)
+        } else {
+            QuicClientCertificate::new(&Keypair::new())
```

Comment:
I think that this can only happen when called from the test ::create_client
method. If that's the case, instead of leaking a test detail all the way down to
here, handle the None part in the tests, and make it so that all the methods
that lead to here take &Keypair instead of Option

Reply:
Not only: if client wants to create use unstaked connections, he passes `None` as well.

Reply:
Then please make new take Option, no need to add another method and
"with_option" wouldn't be a good descriptive name